components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -408,4 +423,231 @@ function MapFunctions.getMapOpponentSize(opponent)
 	return Table.size(Array.filter(opponent.players or {}, Logic.isNotEmpty))
 end
 
+---
+--- FFA specific stuff
+---
+
+---@param matchInput table
+---@param options table?
+---@return table
+function CustomMatchGroupInput.processFFAMatch(matchInput, options)
+	matchInput.bestof = tonumber(matchInput.firstto) or tonumber(matchInput.bestof)
+
+	local match = MatchGroupInputUtil.standardProcessFfaMatch(matchInput, FfaMatchFunctions)
+
+	return match
+end
+
+
+---@param match table
+---@param numberOfOpponents integer
+---@return table
+function FfaMatchFunctions.parseSettings(match, numberOfOpponents)
+	return {
+		noscore = not Logic.readBool(match.hasscore),
+		showgamedetails = false,
+	}
+end
+
+---@param opponent table
+---@param opponentIndex integer
+---@param match table
+function FfaMatchFunctions.adjustOpponent(opponent, opponentIndex, match)
+	MatchFunctions.adjustOpponent(opponent, opponentIndex)
+	-- set score to 0 for all opponents if it is a match without scores
+	if not Logic.readBool(match.hasscore) then
+		opponent.score = 0
+	end
+end
+
+---@param opponents table[]
+---@param games table[]
+---@return fun(opponentIndex: integer): integer?
+function FfaMatchFunctions.calculateMatchScore(opponents, games)
+	return function(opponentIndex)
+		local opponent = opponents[opponentIndex]
+		local sum = (opponent.extradata.advantage or 0) - (opponent.extradata.penalty or 0)
+		Array.forEach(games, function(game)
+			local scores = Array.map(game.opponents, Operator.property('score'))
+			sum = sum + ((scores or {})[opponentIndex] or 0)
+		end)
+		return sum
+	end
+end
+
+---@param opponents table[]
+---@return string
+function FfaMatchFunctions.getMode(opponents)
+	return MODE_FFA
+end
+
+---@param match table
+---@param opponents {score: integer?}[]
+---@return boolean
+function FfaMatchFunctions.matchIsFinished(match, opponents)
+	if MatchGroupInputUtil.isNotPlayed(match.winner, match.finished) then
+		return true
+	end
+
+	local finished = Logic.readBoolOrNil(match.finished)
+	if finished ~= nil then
+		return finished
+	end
+
+	-- If a winner has been set
+	if Logic.isNotEmpty(match.winner) then
+		return true
+	end
+
+	-- If enough time has passed since match started, it should be marked as finished
+	local threshold = match.dateexact and ASSUME_FINISHED_AFTER.EXACT or ASSUME_FINISHED_AFTER.ESTIMATE
+	if match.timestamp ~= DateExt.defaultTimestamp and (match.timestamp + threshold) < NOW then
+		return true
+	end
+
+	return MatchFunctions.placementHasBeenSet(opponents)
+end
+
+---@param opponents table[]
+---@return boolean
+function FfaMatchFunctions.placementHasBeenSet(opponents)
+	return Array.any(opponents, function(opponent) return Logic.isNumeric(opponent.placement) end)
```

Comment:
```suggestion
	return Array.all(opponents, function(opponent) return Logic.isNumeric(opponent.placement) end)
```
I would recommend at least. Otherwise one cannot enter result as they come in (eg. last place is known due to them dying)